pkg/restapi/v1/verifier/controller.go

```diff
@@ -552,55 +561,51 @@ func (c *Controller) verifyAuthorizationResponseTokens(
 
 	var processedVPTokens []*oidc4vp.ProcessedVPToken
 
-	for _, vpt := range authResp.VPToken {
-		vpTokenClaims, signer, err := validateVPToken(vpt, c.jwtVerifier)
+	for _, vpToken := range authResp.VPToken {
+		var vpTokenClaims *VPTokenClaims
+
+		vpTokenClaims, err = c.validateRawVPToken(vpToken)
 		if err != nil {
 			return nil, err
 		}
 
 		logger.Debugc(ctx, "CheckAuthorizationResponse vp_token verified")
 
+		//todo: consider to apply this validation for JWT VP in verifypresentation.Service
 		if vpTokenClaims.Nonce != idTokenClaims.Nonce {
 			return nil, resterr.NewValidationError(resterr.InvalidValue, "nonce",
 				errors.New("nonce should be the same for both id_token and vp_token"))
 		}
 
-		presentation, err := verifiable.ParsePresentation(vpTokenClaims.VP,
-			verifiable.WithPresPublicKeyFetcher(
-				verifiable.NewVDRKeyResolver(c.vdr).PublicKeyFetcher(),
-			),
-			verifiable.WithPresJSONLDDocumentLoader(c.documentLoader),
-		)
-		if err != nil {
-			return nil, resterr.NewValidationError(resterr.InvalidValue, "vp_token.vp", err)
+		if vpTokenClaims.Aud != idTokenClaims.Aud {
```

Comment:
Along with `vpTokenClaims.Nonce != idTokenClaims.Nonce` check I added `vpTokenClaims.Aud != idTokenClaims.Aud` check, accodring to [SIOP Authentication Response example from doc](https://openid.net/specs/openid-connect-4-verifiable-presentations-1_0-ID1.html#name-self-issued-openid-provider).